components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -3794,6 +3795,31 @@ public void updatePermissionListOfGroup(String groupName, String[] permissionToA
         return userSchemaAttributesList;
     }
 
+    /**
+     * Returns the schema of the enterprise user extension in SCIM 2.0.
+     * @return List of attributes of enterprise user extension
+     * @throws CharonException
+     */
+    @Override
+    public List<Attribute> getEnterpriseUserSchema() throws CharonException {
+
+        List<Attribute> enterpriseUserSchemaAttributesList = null;
+
+        if (SCIMCommonUtils.isEnterpriseUserExtensionEnabled()) {
+            Map<ExternalClaim, LocalClaim> scimClaimToLocalClaimMap =
+                    getMappedLocalClaimsForDialect(SCIMCommonConstants.SCIM_ENTERPRISE_USER_CLAIM_DIALECT, tenantDomain);
+
+            Map<String, Attribute> filteredAttributeMap =
+                    getFilteredEnterpriseUserSchemaAttributes(scimClaimToLocalClaimMap);
+            Map<String, Attribute> hierarchicalAttributeMap =
+                    buildHierarchicalAttributeMap(filteredAttributeMap);
+
+            enterpriseUserSchemaAttributesList = new ArrayList(hierarchicalAttributeMap.values());
```

Comment:
shall we add a debug log to log the attributes in enterpriseuserschema similar to             logSchemaAttributes(userSchemaAttributesList);